src/tests/unit/protocols/dhcpv6/rfc6355.txt

```diff
@@ -45,6 +45,26 @@ match 00 01 00 02 00 04
 decode-pair -
 match Client-ID-DUID = Client-ID-DUID-UUID
 
+#  Encode a 16-octet UUID
+encode-pair Client-ID-DUID = Client-ID-DUID-UUID, Client-ID-DUID-UUID-Value = 0x000102030405060708090a0b0c0d0e0f
+match 00 01 00 12 00 04 00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f
+
+#  UUID-Value field is 16 octets.  So we truncate if if it's too long, and pad it with zeros if it's too short.
+encode-pair Client-ID-DUID = Client-ID-DUID-UUID, Client-ID-DUID-UUID-Value = 0x000102030405060708090a0b0c0d0e0f1112
+match 00 01 00 12 00 04 00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f
+
+#  But if we decode something that's too long, we ignore the extra data
+decode-pair -
+match Client-ID-DUID = Client-ID-DUID-UUID, Client-ID-DUID-UUID-Value = 0x000102030405060708090a0b0c0d0e0f
+
+#  Or we pad the UUID to 16 octets if it's too short
+encode-pair Client-ID-DUID = Client-ID-DUID-UUID, Client-ID-DUID-UUID-Value = 0x000102030405060708090a0b0c0d
+match 00 01 00 12 00 04 00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 00 00
```

Comment:
If the UUID is expected to be 16 octets and it's shorter, then it's malformed and must be created as an unknown attribute

Reply:
Unless the RFC specifically mentions different behaviour?

Reply:
This is encoding, not decoding.  So we can't create an unknown attribute in the packet.

The RFC says that the field is 16 octets.  If the admin sets a UUID which is shorter, then the encoder pads out the field to 16 octets.